Make iOS null behavior of SharedPreferences not crash

collinjackson opened the pull request:
Do what we do on Android and clear the value instead.

Fixes flutter/flutter#11774

goderbauer commented on packages/shared_preferences/ios/Classes/SharedPreferencesPlugin.m:
```diff
@@ -20,6 +20,10 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
+      if ([[NSNull null] isEqual:value])
+        [[NSUserDefaults standardUserDefaults] removeObjectForKey:key];
+      else
+        [[NSUserDefaults standardUserDefaults] setValue:value forKey:key];
```
Should this be `setBool` instead of `setValue`?

collinjackson replied:
I had meant to use the line below instead

goderbauer commented on packages/shared_preferences/ios/Classes/SharedPreferencesPlugin.m:
```diff
@@ -20,6 +20,10 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
+        [[NSUserDefaults standardUserDefaults] removeObjectForKey:key];
+      else
+        [[NSUserDefaults standardUserDefaults] setValue:value forKey:key];
       [[NSUserDefaults standardUserDefaults] setBool:value.boolValue forKey:key];
```
Remove this line?

collinjackson replied:
Good catch, thanks

goderbauer commented:
Can you also add something to the changelog and bump the version?

collinjackson commented:
Added crash fix for Android as well, PTAL

goderbauer commented on packages/shared_preferences/lib/shared_preferences.dart:
```diff
@@ -84,10 +84,16 @@ class SharedPreferences {
   void _setValue(String valueType, String key, Object value) {
```
Would probably be cool to document that `null` removes a key in the doc comments of the methods.

collinjackson replied:
Good idea

collinjackson commented:
After some reflect, I think it's reasonable to have `remove()` be part of the public Dart API, since both iOS and Android have a native equivalent of that. I rewrote the patch to reflect this and added a test -- let me know if this looks good.
